src/scenarios/shared/runner.py

```diff
@@ -9,38 +9,69 @@
 from collections import namedtuple
 from argparse import ArgumentParser
 from shared.startup import StartupWrapper
-from shared.util import publishedexe, extension
+from shared.util import publishedexe, extension, pythoncommand, iswin
 from shared import const
 from performance.logger import setup_loggers
 
-
-reqfields = ('exename',
-            )
-optfields = ('guiapp',
-             'startupmetric',
-             'appargs',
-             'environmentvariables',
-             'iterations',
-             'timeout',
-             'warmup',
-             'workingdir',
-             'iterationsetup',
-             'setupargs',
-             'iterationcleanup',
-             'cleanupargs',
-             'processwillexit',
-             'measurementdelay'
-             )
-
 # These are the kinds of scenarios we run. Default here indicates whether ALL
 # scenarios should try and run a given test type.
 testtypes = {const.STARTUP: False,
              const.SDK: False,
              const.CROSSGEN: False}
 
-TestTraits = namedtuple('TestTraits',
-                        reqfields  + tuple(testtypes.keys()) + optfields,
-                        defaults=tuple(testtypes.values()) + (None,) * len(optfields))
+
+class TestTraits:
```

Comment:
Should this go into its own file now? When it was just a tuple it seemed reasonable to have here but now that it has code etc maybe it should move?

Reply:
Yeah I also prefer having it in another file.